src/app/query/bitcoin/address/utxos-by-address.hooks.ts

```diff
@@ -1,48 +1,57 @@
 import { useCallback } from 'react';
 
+import { InscriptionResponseItem } from '@shared/models/inscription.model';
+
 import { useCurrentAccountNativeSegwitIndexZeroSigner } from '@app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks';
 
 import { UtxoResponseItem } from '../bitcoin-client';
-import { useInscriptionByAddressQuery } from '../ordinals/use-inscriptions.query';
+import { useInscriptionsByAddressQuery } from '../ordinals/inscriptions.query';
 import { useBitcoinPendingTransactionsInputs } from './transactions-by-address.hooks';
 import { useGetUtxosByAddressQuery } from './utxos-by-address.query';
 
+export function filterUtxosWithInscriptions(
+  inscriptions: InscriptionResponseItem[],
+  utxos: UtxoResponseItem[]
+) {
+  return utxos.filter(
+    utxo =>
+      !inscriptions?.some(
+        inscription => `${utxo.txid}:${utxo.vout.toString()}` === inscription.output
+      )
+  );
+}
+
 /**
  * Warning: ⚠️ These are **all** UTXOs, including Stamped and Inscribed UTXOs.
- * You should probably use `useSpendableCurrentNativeSegwitAccountUtxos` instead.
+ * You should probably use `useCurrentNativeSegwitAccountSpendableUtxos` instead.
  */
 export function useCurrentNativeSegwitUtxos() {
   const nativeSegwitSigner = useCurrentAccountNativeSegwitIndexZeroSigner();
   return useGetUtxosByAddressQuery(nativeSegwitSigner.address);
 }
 
-function useFilterAddressNativeSegwitInscriptions(address: string) {
+function useFilterInscriptionsByAddress(address: string) {
```

Comment:
I renamed a few of these hooks, removing the `NativeSegwit` portion, bc they really just accept any address. They could be reused for taproot addresses. I left the ns in the name when it is actually using the ns address in the hook (at the end of the hook chain).